packages/common.vm/tools/vm.common/vm.common.psm1

```diff
@@ -503,6 +503,110 @@ function VM-Remove-Tool-Shortcut {
+        # $silentArgs = '-s -u'          # Ghost
+        [string] $silentArgs,
+        [Parameter(Mandatory=$true, Position=4)]
+        [string] $executablePath,
+        [Parameter(Mandatory=$true, Position=5)]
+        [string] $url,
+        [Parameter(Mandatory=$false)]
+        [string] $sha256,
+        [Parameter(Mandatory=$false)]
+        [array] $validExitCodes= @(0, 3010, 1605, 1614, 1641),
+        [Parameter(Mandatory=$false)]
+        [bool] $consoleApp=$false
+    )
+    try {
+        $toolDir = Join-Path ${Env:RAW_TOOLS_DIR} $toolName
+
+        # Get the file extension from the URL
+        $installerName = Split-Path -Path $url -Leaf
+        $ext = $installerName.Split(".")[-1].ToLower()
+
+        # Download and install
+        $packageArgs = @{
+            packageName   = ${Env:ChocolateyPackageName}
+            url           = $url
+            checksum      = $sha256
+            checksumType  = "sha256"
+        }
+        if ($ext -in @("zip", "7z")) {
+            VM-Remove-PreviousZipPackage ${Env:chocolateyPackageFolder}
+            $unzippedDir= Join-Path $toolDir "$($toolName)_installer"
+            $packageArgs['unzipLocation'] = $unzippedDir
+            Install-ChocolateyZipPackage @packageArgs
+            VM-Assert-Path $unzippedDir
+
+            $exePaths = Get-ChildItem $unzippedDir | Where-Object { $_.Name.ToLower() -match '^.*\.(exe|msi)$' }
+            if ($exePaths.Count -eq 1) {
+                $installerPath = $exePaths[0].FullName
+            } else {
+                $exePaths = Get-ChildItem $unzippedDir | Where-Object { $_.Name.ToLower() -match '^.*(setup|install).*\.(exe|msi)$' }
+                if ($exePaths.Count -eq 1) {
+                    $installerPath = $exePaths[0].FullName
+                } else {
+                    throw "Unable to determine installer file within: $unzippedDir"
+                }
+            }
+        } else {
+            $installerPath = Join-Path $toolDir $installerName
+            $packageArgs['fileFullPath'] = $installerPath
+            Get-ChocolateyWebFile @packageArgs
+            VM-Assert-Path $installerPath
+        }
+
+        # Install tool via native installer
+        $packageArgs = @{
+            packageName   = ${Env:ChocolateyPackageName}
+            fileType      = $fileType
+            file          = $installerPath
+            silentArgs    = $silentArgs
+            validExitCodes= $validExitCodes
+            softwareName  = $toolName
+        }
+        Install-ChocolateyInstallPackage @packageArgs
+        VM-Assert-Path $executablePath
+
+        $shortcutDir = Join-Path ${Env:TOOL_LIST_DIR} $category
+        $shortcut = Join-Path $shortcutDir "$toolName.lnk"
+        if ($consoleApp) {
+            $executableCmd  = Join-Path ${Env:WinDir} "system32\cmd.exe"
+            $executableDir  = Join-Path ${Env:UserProfile} "Desktop"
+            $executableArgs = "/K `"cd `"$executableDir`" && `"$executablePath`" --help`""
```

Comment:
Will `--help` always work or could it cause confusion/issues?

Reply:
Great question 😄 It may not always work, but at the moment I don't have better solution besides removing it. The idea was to try to display help information rather than running the program without input. If this causes issues in the future, we can address it then?

Reply:
Yes, sure! Let's wait for an actual issues first.